lib/scnnr/connection.rb

```diff
@@ -46,7 +46,7 @@ def send_request
 
     def with_retries
       yield
-    rescue Errno::ECONNRESET => e
+    rescue Errno::ECONNRESET, Timeout::Error => e
```

Comment:
Added this so we can rescue `Net::ReadTimeout` and `Net::OpenTimeout` (they inherit after `Timeout::Error`). Following same pattern I was considering changing `Errno::ECONNRESET` to `SystemCallError` to handle more `Errno::` errors.
@hiroara what do you think?

References:
https://airbrake.io/blog/ruby-exception-handling/systemcallerror
https://www.ibm.com/support/knowledgecenter/en/SS6NHC/com.ibm.swg.im.dashdb.messages.doc/doc/r0058740.html

Reply:
Thank you for improving it!
About `SystemCallError`, I think it's too wide because many errors unrelated to networking could inherit that error type.